pkg/authorization/api/v1/types.go

```diff
@@ -176,6 +176,10 @@ type SubjectAccessReview struct {
 	User string `json:"user"`
 	// GroupsSlice is optional. Groups is the list of groups to which the User belongs.
 	GroupsSlice []string `json:"groups"`
+	// Scopes to use for the evaluation.  Empty means "use the unscoped (full) permissions of the user/groups".
+	// Nil for a self-SAR, means "use the scopes on this request".
+	// Nil for a regular SAR, means the same as empty.
+	Scopes []string `json:"scopes,omitempty"`
```

Comment:
If you add omitempty here then you will always get nil (even if someone provides empty).  So there's no point to describing nil and empty differently.  If you didn't want that behavior, you should remove omitempty.